packages/klarna-integration/src/klarnav2/KlarnaV2PaymentMethod.spec.tsx

```diff
@@ -0,0 +1,149 @@
+
+        defaultProps = {
+            method,
+            checkoutService,
+            checkoutState,
+            paymentForm: getPaymentFormServiceMock(),
+            language: createLanguageService(),
+            onUnhandledError: jest.fn(),
+        };
+
+        PaymentMethodTest = (props) => (
+            <CheckoutProvider checkoutService={checkoutService}>
+                <LocaleContext.Provider value={localeContext}>
+                    <Formik initialValues={{}} onSubmit={noop}>
+                        <KlarnaV2PaymentMethod {...props} />
+                    </Formik>
+                </LocaleContext.Provider>
+            </CheckoutProvider>
+        );
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('renders as hosted widget component', () => {
+        const container = mount(<PaymentMethodTest {...defaultProps} method={method} />);
+        const component = container.find(HostedWidgetPaymentComponent);
+
+        expect(component.props()).toEqual(
+            expect.objectContaining({
+                containerId: `${method.id}Widget`,
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                deinitializePayment: expect.any(Function),
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                initializePayment: expect.any(Function),
+                method,
+            }),
+        );
+    });
+
+    it('initializes method with required config when no instruments', () => {
+        jest.spyOn(checkoutState.data, 'getInstruments').mockReturnValue(undefined);
+
+        const container = mount(<PaymentMethodTest {...defaultProps} method={method} />);
+        const component = container.find(HostedWidgetPaymentComponent);
+
+        expect(component.props()).toEqual(
+            expect.objectContaining({
+                containerId: `${method.id}Widget`,
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                deinitializePayment: expect.any(Function),
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                initializePayment: expect.any(Function),
+                method,
+            }),
+        );
+    });
+
+    it('initializes method with required config when customer is defined', () => {
+        jest.spyOn(checkoutState.data, 'getCustomer').mockReturnValue(getCustomer());
+
+        const container = mount(<PaymentMethodTest {...defaultProps} method={method} />);
+        const component = container.find(HostedWidgetPaymentComponent);
+
+        expect(component.props()).toEqual(
+            expect.objectContaining({
+                containerId: `${method.id}Widget`,
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                deinitializePayment: expect.any(Function),
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```

Comment:
Its ok to turn off this eslint rule for whole package for spec and test files. Also, as far as I remember we should use `.test.tsx` instead of `.spec.tsx`

Reply:
I will set these rules for whole package and change extension from .spec.tsx to .test.tsx